stock_quant_manual_assign/tests/test_stock_quant_manual_assign.py

```diff
@@ -172,3 +174,23 @@ def test_quant_assign_wizard_after_availability_check(self):
             sum(line.qty for line in wizard.quants_lines),
             self.move.reserved_availability,
         )
+
+    def test_quant_assign_get_discrepancies(self):
+        self.move._action_assign()
+        logging.info(self.move.move_line_ids)
+        # There should be no move lines to unlink and no quant lines to assign
+        # if there is no discrepancy between move lines and quant lines
+        wizard = self._create_wizard()
+        move_lines_to_unlink, quant_lines_to_assign = wizard._get_discrepancies()
+        self.assertFalse(move_lines_to_unlink)
+        self.assertFalse(quant_lines_to_assign)
+        # There should be a move line to unlink (due to owner mismatch)
+        # and a quant line to assisgn (due to qty discrepancy)
+        owner = self.env["res.partner"].create({"name": "Owner"})
+        self.move.move_line_ids[0].owner_id = owner.id
+        wizard = self._create_wizard()
+        wizard.quants_lines[0].write({"selected": True, "qty": 100.0})
+        wizard.quants_lines[1].write({"selected": True, "qty": 100.0})
```

Comment:
The scenario is still hard to follow. IIRC, `self._create_wizard()` creates three selected quant line records (with no owner, since the value is taken from the quant), which creates a discrepancy for the first quant line, since the corresponding move line has an owner assigned in the previous step.

If this is the case, we shouldn't need the steps of selecting quants. Instead, we should just assert the contents of `self.move.move_line_ids` and `wizard.quants_lines`. Am I missing something?

Reply:
Please review my latest changes.